### ISSUE TITLE
Refactor `avatar` to make sure it never fails to generate a valid hex.

### ISSUE THREAD

**Issue by Elijbet** [Esri/calcite-design-system#6593]
### Description

`private generateFillColor` function should never generate an invalid hex

### Proposed Advantages

Simplify code to prevent errors in coordinating background and foreground colors, as this is important for validating contrast. 

### Which Component

avatar

### Relevant Info

Stems from https://github.com/Esri/calcite-components/pull/6528 - refer to [this comment](https://github.com/Esri/calcite-design-system/pull/6528#issuecomment-1463083721) for specifics in the PR.

**Comment by jcfranco** [comment]
@Elijbet Is this still applicable?

**Comment by Elijbet** [comment]
> @Elijbet Is this still applicable?

I think the description doesn't provide sufficient context tbh. The contrast issue has been addressed in multiple PRs and has been verified here: https://github.com/Esri/calcite-design-system/issues/6203#issuecomment-1470173704, so I'd just close it.
